libgstd/libgstd.c

```diff
@@ -0,0 +1,232 @@
+
+  return code_description[code];
+}
+
+static void
+gstd_manager_init (int argc, char *argv[])
+{
+  gst_init (&argc, &argv);
+  gstd_debug_init ();
+}
+
+gboolean
+gstd_manager_start (GstDManager * manager)
+{
+  gboolean ipc_selected = FALSE;
+  gboolean ret = TRUE;
+  GstdReturnCode code = GSTD_EOK;
+  gint ipc_idx;
+
+  gstd_assert_and_ret_val (NULL != manager, GSTD_NULL_ARGUMENT);
+  gstd_assert_and_ret_val (NULL != manager->ipc_array, GSTD_LIB_NOT_FOUND);
+  gstd_assert_and_ret_val (NULL != manager->session, GSTD_LIB_NOT_FOUND);
+
+  /* Verify if at least one IPC mechanism was selected */
+  for (ipc_idx = 0; ipc_idx < manager->num_ipcs; ipc_idx++) {
+    g_object_get (G_OBJECT (manager->ipc_array[ipc_idx]), "enabled",
+        &ipc_selected, NULL);
+
+    if (ipc_selected) {
+      break;
+    }
+  }
+
+  /* If no IPC was selected, default to TCP */
+  if (!ipc_selected) {
+    g_object_set (G_OBJECT (manager->ipc_array[0]), "enabled", TRUE, NULL);
+  }
+
+  /* Run start for each IPC (each start method checks for the enabled flag) */
+  for (ipc_idx = 0; ipc_idx < manager->num_ipcs; ipc_idx++) {
+    code = gstd_ipc_start (manager->ipc_array[ipc_idx], manager->session);
+    if (code) {
+      g_printerr ("Couldn't start IPC : (%s)\n",
+          G_OBJECT_TYPE_NAME (manager->ipc_array[ipc_idx]));
+      ret = FALSE;
+    }
+  }
+
+  return ret;
+}
+
+static void
+gstd_manager_ipc_stop (GstDManager * manager)
+{
+  gstd_assert_and_ret (NULL != manager);
+  gstd_assert_and_ret (NULL != manager->ipc_array);
+  gstd_assert_and_ret (NULL != manager->session);
+
+  /* Run stop for each IPC */
+  for (gint ipc_idx = 0; ipc_idx < manager->num_ipcs; ipc_idx++) {
+    if (NULL != manager->ipc_array[ipc_idx]
+        && TRUE == manager->ipc_array[ipc_idx]->enabled) {
+      gstd_ipc_stop (manager->ipc_array[ipc_idx]);
+      g_clear_object (&manager->ipc_array[ipc_idx]);
+    }
+  }
+}
+
+static void
+gstd_manager_set_ipc (GstDManager * manager)
+{
+  /* Array to specify gstd how many IPCs are supported, 
+   * SupportedIpcs should be added to this array.
+   */
+  const SupportedIpcs supported_ipcs[] = {
+    GSTD_IPC_TYPE_TCP,
+    GSTD_IPC_TYPE_UNIX,
+    GSTD_IPC_TYPE_HTTP,
+  };
+
+  const guint num_ipcs = (sizeof (supported_ipcs) / sizeof (SupportedIpcs));
+
+  GstdIpc **ipc_array = NULL;
+
+  gstd_assert_and_ret (NULL != manager);
+  gstd_assert_and_ret (NULL != supported_ipcs);
+
+  /* If there is ipcs, then initialize them */
+  if (NULL != supported_ipcs && num_ipcs > 0) {
+    ipc_array = g_malloc0 (num_ipcs * sizeof (*ipc_array));
+    for (gint ipc_idx = 0; ipc_idx < num_ipcs; ipc_idx++) {
+      ipc_array[ipc_idx] =
+          GSTD_IPC (g_object_new (gstd_supported_ipc_to_ipc (supported_ipcs
+                  [ipc_idx]), NULL));
+    }
+    manager->ipc_array = ipc_array;
+  }
+
+  manager->num_ipcs = num_ipcs;
+  manager->ipc_array = ipc_array;
+}
+
+void
+gstd_get_option_context (GstDManager * manager, GOptionContext ** context)
```

Comment:
Why do we receive a double pointer? Double pointers should be used for output parameters that are going to be allocated in the function. This is not the case here, the context comes from the outside. This should be a single pointer.